### DISCUSSION TITLE
Legality of Docker-OSX

### DISCUSSION THREAD

**Discussion by TheScientist101** [sickcodes/Docker-OSX#267] [Q&A]
Is using this docker image legal, or is it like a hackintosh vm?

**Comment by sickcodes** [comment]
I use Docker-OSX for security research and I am currently undergoing responsible disclosure with one major vulnerability, so yes, it is legal:

https://sick.codes/is-hackintosh-osx-kvm-or-docker-osx-legal/

**Reply by TheScientist101** [reply]
So, if I was to use it for making iPhone Apps in Xcode, would that be considered legal? According to the article, I don't think so, but I just want to double-check.

**Reply by IARI** [reply]
Same question!
We develop an app for android and iOS and I would really love to use this image for our CI/CD process if possible.

**Reply by bcheidemann** [reply]
Ditto

**Comment by avijoni** [comment]
As far as I know EULA licenses don't apply in EU. At least in some countries, since by EU consumer legislation you can use the product you bought any way you like ;) So if you have apple computer you are allowed to use their OS in any way you want. You paid for it when you bought your mac. So technically it's not illegal. I know this, because there used to be company in Germany that sold hackintosh-es legally. 
https://arstechnica.com/gadgets/2009/02/german-company-pearc-begins-selling-line-of-mac-clones-1/

**Reply by avijoni** [reply]
but i just realized that back in the days, Apple was selling OSX separately, so I guess it's not that legal anymore, because you can't buy macos separately anymore. Ignore my previous comment.

**Reply by bcheidemann** [reply]
So I have no idea how this works but if I buy a car, I can legally remove the tyres and use them on a different car. Hypothetically, if I created this docker image from my own mac, would this be any different?

**Comment by om-ha** [comment]
Docker-OSX has HUGE potential for CI/CD purposes...


> Developers use it to automate releases of their Applications, minimizing human error, and maximizing end-user security.
[source](https://sick.codes/is-hackintosh-osx-kvm-or-docker-osx-legal)


1. Is it legal for **individuals** to use this for CI/CD?
2. Is it legal for **companies** to use this for CI/CD?

**Reply by om-ha** [reply]
The individual/company has already paid for MacBooks, you're just extracting the image of the device YOU purchased. Albeit you are making tens of workers of the same image, so it's not a one-to-one relation, it's more like one-to-many.
